src/app/client/src/app/modules/content-search/components/view-all/view-all.component.ts

```diff
@@ -211,17 +211,32 @@ export class ViewAllComponent implements OnInit, OnDestroy, AfterViewInit {
   getContents(data) {
     this.getContentList(data).subscribe((response: any) => {
       this.showLoader = false;
-      if (response.contentData.result.count && response.contentData.result.content) {
-        this.noResult = false;
-        this.totalCount = response.contentData.result.count;
-        this.pager = this.paginationService.getPager(response.contentData.result.count, this.pageNumber, this.pageLimit);
-        this.searchList = this.formatSearchresults(response);
+      if (this.sectionName === this.resourceService.frmelmnts.lbl.mytrainings) {
+        if (_.get(response, 'enrolledCourseData.enrolledCourses')) {
+          const enrolledCourseCount = _.get(response, 'enrolledCourseData.enrolledCourses').length;
+          this.noResult = false;
+          this.totalCount = enrolledCourseCount;
+          this.searchList = this.formatSearchresults(_.orderBy(_.get(response, 'enrolledCourseData.enrolledCourses'), ['enrolledDate'], ['desc']));
+        } else {
+          this.noResult = true;
```

Comment:
Add this into another function & call the function from here.

Reply:
Both if-else condition can be added into another function.

Reply:
only changing this for my courses section. moving to another function.